text/0015-expose-handeled-property.md

```diff
@@ -0,0 +1,43 @@
+* Start Date: 2022-09-26
+* RFC Type: feature
+* RFC PR: https://github.com/getsentry/rfcs/pull/15
+
+# Summary
+
+Allow a user to decide if an exception/error is considered `handled: true` or `handled: false`.
+
+# Motivation
+
+A common way to integrate Sentry into a PHP framework is to utilize the framework's error handler stack.
+
+```php
+use Psr\Http\Message\ServerRequestInterface;
+use Throwable;
+use function Sentry\captureException;
+
+public function logException(
+    Throwable $exception,
+    ?ServerRequestInterface $request = null,
+    bool $includeTrace = false
+): void {
+    captureException($exception);
```

Comment:
What most SDKs do here is to create a new SentryEvent, with a mechanism and handled=false and instead of calling `captureException`, they call `captureEvent`, is that even possible?

Reply:
Example how Python does this: https://github.com/getsentry/sentry-python/blob/master/sentry_sdk/integrations/flask.py#L204-L219

Reply:
`Client::captureException()` calls `Client::captureEvent(Event::createEvent(), $hint, $scope);` in PHP, but attaches the `Exception` to the `hint`. Still, I can't access `handled` here.
Maye this whole RFC is just trying to fix an issue in the PHP SDK, idk.

Reply:
Sounds like that, the `mechanism` is part of the event protocol and should be accessible, if that's possible to fix, I'd go this way instead of exposing new public APIs.

Reply:
I'm also +1 on exposing the mechanism to users - but then we should make a decision on what `handled` does in https://github.com/getsentry/rfcs/pull/10 first.

Reply:
Makes sense, but I'd not block this RFC because of #10.
My recommendation would be: simply do `handled: false` if it's an unhandled issue and it should be shown as such in the UI + being able to filter and alert on them.